include/OpenColorIO/OpenColorIO.h

```diff
@@ -3437,6 +3561,60 @@ class OCIOEXPORT SystemMonitors
     virtual ~SystemMonitors() = default;
 };
 
+
+///////////////////////////////////////////////////////////////////////////
+// ConfigIOProxy
+
+/**
+ * ConfigIOProxy is a proxy class to allow the calling program to supply the config and any
+ * associated LUT files directly, without relying on the standard file system. 
+ * 
+ * The OCIOZ archive feature is implemented using this mechanism.
+ */
+class OCIOEXPORT ConfigIOProxy
+{
+public:
+    ConfigIOProxy() = default;
+    virtual ~ConfigIOProxy() = default;
+
+    /**
+     * \brief Provide the contents of a LUT file as a buffer of uint8_t data.
+     * 
+     * \param buffer Contents of the LUT file.
+     * \param filepath Fully resolved path to the "file."
+     * 
+     * The file path is based on the Config's current working directory and is the same absolute
+     * path that would have been provided to the file system.
+     */
+    virtual std::vector<uint8_t> getLutData(const char * filepath) const = 0;
+
+    /**
+     * \brief Provide the config file Yaml to be parsed.
+     * 
+     * \return String with the config Yaml.
+     */
+    virtual const std::string getConfigData() const = 0;
```

Comment:
Once the config string or LUT data is returned, it shouldn't be editable. Is there a scenario where this should be possible?

Reply:
Usually as we return by value it doesn't matter to be const, that's why you almost never see return by const val I guess (or at least I don't see it!). Because it's already a copy, it's not really worth imposing constraint on the caller I would say.

Reply:
My initial intention was to let the compilers optimize with Copy Elision or NRVO (based on my understanding), but now I realize the const here will prevent any move semantics.

The string shouldn't be copied here because of compiler optimization, but you are right that returning by const value doesn't make sense.

Reply:
There are cases where a compiler is required by the standard to perform copy elision https://en.cppreference.com/w/cpp/language/copy_elision could check if we fall into the listed cases. But yeah removing the const is the way to go since C++11 I think.